File: homework_2/concatenate.c
#include <stdio.h>
#include <math.h>

int concatenate(int num1, int num2) {
	int nOfDigits2 = 0; //variable for storing the number of digits of num2
	
	int temp = num2; //temporary variable to use in the loop

	while(temp > 0) {
		nOfDigits2++; 
		temp /= 10; //increase the number of digits and remove last digit of temp
	}

	num1 = num1 * pow(10, nOfDigits2); 

	int result = num1 + num2;

	return result;
}

int main() {
	int number1, number2; //declare the input variables

	printf("Enter two positive integers: ");
	scanf("%d %d", &number1, &number2); //get user input and assign it to previously declared variables
	
	int result = concatenate(number1, number2); //call the "concatenate" function 

	printf("The concatenated number is %d \n", result); //print the result
	return 0;
}

File: homework_2/concatenate.py
def concatenate(number1, number2):
    result = 0
    
    nOfDigits = 0
    temp = number2
    
    while temp > 0:
        nOfDigits += 1 
        temp //= 10
        
    result = number1 * 10**nOfDigits + number2
    
    return result

def main():
    number1 = int(input("Enter a number: "))
    number2 = int(input("Enter a number: "))
    
    result = concatenate(number1, number2)
    
    print("The concatenated number is", result)
    return 0
    
if __name__ == "__main__":
    main()    
